keras_cv/models/resnet_v1.py

```diff
@@ -141,23 +200,32 @@ def apply(x):
     return apply
 
 
-def Stack(filters, blocks, stride=2, name=None):
+def Stack(
+    filters, blocks, stride=2, name=None, block_type="block", first_shortcut=True
+):
     """A set of stacked residual blocks.
     Args:
-      filters: integer, filters of the bottleneck layer in a block.
+      filters: integer, filters of the layers in a block.
       blocks: integer, blocks in the stacked blocks.
       stride1: default 2, stride of the first layer in the first block.
       name: string, stack label.
+      block_type: string, "block" or "basic_block".
+      first_shortcut: default True, use convolution shortcut if True,
+          otherwise identity shortcut.
     Returns:
       Output tensor for the stacked blocks.
     """
     if name is None:
         name = f"v1_stack_{backend.get_uid('v1_stack')}"
 
+    block_fn = Block if block_type == "block" else BasicBlock
```

Comment:
lets change this to just accept block_fn as an argument instead of block_type.

Then you can write Stack(block_fn=BasicBlock)

Reply:
alternatively explicitly iterate the acceptable parameters for block_type

Reply:
sure, corrected that!